File: AtecaApp/AtecaAPI/Models/Entity/Reserva.cs
namespace AtecaAPI.Models.Entity
{
    public class Reserva
    {
        public int Id { get; set; }

        public DateOnly Fecha { get; set; }


        public int FranjaHorariaId { get; set; }
        public FranjaHoraria FranjaHoraria { get; set; } = null!;

        public int ProfesorId { get; set; }
        public Profesor Profesor { get; set; } = null!;

        public int GrupoClaseId { get; set; }
        public GrupoClase GrupoClase { get; set; } = null!;

        public string Estado { get; set; } = "Pendiente";

        public DateTime FechaSolicitud { get; set; } = DateTime.Now;
        public DateTime? FechaResolucion { get; set; }
    }


}

File: AtecaApp/AtecaApp_Angular/src/app/models/reserva.ts
// src/app/models/reserva.ts
import { FranjaHoraria } from './franjaHoraria';
import { GrupoClase } from './grupoClase';
import { Profesor } from './profesor';



export interface Reserva {
  id: number;
  profesor: Profesor;
  grupoClase: GrupoClase;
  fecha: string;
  franjaHoraria: FranjaHoraria;
  estado: string;
  fechaSolicitud: string; 
  fechaResolucion?: string | null;
}
